engine/tests/phpunit/unit/ElggPluginUnitTest.php

```diff
@@ -93,13 +93,21 @@ public function testCanLoadStaticConfig() {
 					'path' => '/plugin/{foo?}',
 					'resource' => 'plugin/foo',
 				],
-			]
+			],
+			'settings' => [
+				'default1' => 'set1',
+			],
+			'user_settings' => [
+				'user_default1' => 'set1',
+			],
 		];
 
 		$this->assertEquals($config['entities'], $plugin->getStaticConfig('entities'));
 		$this->assertEquals($config['actions'], $plugin->getStaticConfig('actions'));
 		$this->assertEquals($config['widgets'], $plugin->getStaticConfig('widgets'));
 		$this->assertEquals($config['routes'], $plugin->getStaticConfig('routes'));
+		$this->assertEquals($config['settings'], $plugin->getStaticConfig('settings'));
+		$this->assertEquals($config['user_settings'], $plugin->getStaticConfig('user_settings'));
```

Comment:
why no checks for `hooks` and `events`

Reply:
they are not present in the test file... just added these as they were not tested... testing reading more sections makes no sense